test/utils/wabt_engine.cpp

```diff
@@ -44,6 +57,26 @@ bool WabtEngine::parse(bytes_view input) const
 
 bool WabtEngine::instantiate(bytes_view wasm_binary)
 {
+    wabt::interp::HostModule* hostModule = m_env.AppendHostModule("env");
+    assert(hostModule != nullptr);
+
+    hostModule->AppendFuncExport("adler32", {{wabt::Type::I32, wabt::Type::I32}, {wabt::Type::I32}},
+        [=](const wabt::interp::HostFunc*, const wabt::interp::FuncSignature*,
```

Comment:
Quite big for a lambda, can you make the code inside a separate private method?

Also why not use the same function in `parse`?

Reply:
Parse doesn't need it, it only wants the signature.

I had a different version using a `uniqu_ptr` but something wasnt working. The problem is `GetMemory` will only return correctly after instantiation, so it can't be captured prior. Also wabt has no context argument, so we'll need a lambda in any case, but could extarct out into a helper.

I would prefer however not tod any of that here, but someone else could optimize these in subsequent PRs.